### PR TITLE
694/cancel order on invalid status

### LINKED ISSUE #694: Return API error when cancelling a finalized order.
If an order gets cancelled after it has already been previously cancelled, or already completely filled, the orderbook API should return an HTTP error (400?).

There might be some funny edge cases that require extra attention (cancelled and then filled because tx with settlement was already getting mined).

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Fixes #694 

Depends on #715 

Adds 3 new error types for order cancellation based on order status.

### Test Plan
Unit tests

### CODE REVIEW

File: orderbook/openapi.yml [hunk 1]
```diff
@@ -588,7 +588,7 @@ components:
       properties:
         errorType:
           type: string
-          enum: [InvalidSignature, WrongOwner, OrderNotFound]
+          enum: [InvalidSignature, WrongOwner, OrderNotFound, OrderAlreadyCancelled, OrderFullyExecuted, OrderExpired]
```
Comment (bh2smith) [inline orderbook/openapi.yml:591]: ubernit: I think `AlreadyCancelled` can live without the term Order in front, but the other two (for some reason) make sense to have... I'm weird.

**Review by bh2smith — APPROVED** [review]
This looks good, but I think the inclusion of partial orders might wind up changing things a bit here. Keep in mind that this won't entirely solve the issues we are hoping for with inflight order cancellation, but it is already much better than we had before. 

Just to elaborate, some of these statuses will depend on our event listener which waits for trade events and updates some of the fields used in the status calculation. 

In any it is an improvement! Nice work.

**Review by fleupold — APPROVED** [review]
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia0.giphy.com%2Fmedia%2FW0Vz7zDK7vgMBc2fpw%2Fgiphy.gif&token=very_nice" data-gitmeme-token="very_nice"></a>

**Review by vkgnosis — APPROVED** [review]

File: orderbook/openapi.yml (hunk 1 above)
Comment (alfetopito) [reply orderbook/openapi.yml:591]: I don't mind either way. Will update.

**Pull request merged**